tests/suites/spaces_ec2/util.sh

```diff
@@ -0,0 +1,99 @@
+ec2_check_prerequisites_for_space_tests() {
+  match_count=$(aws ec2 describe-subnets --filters Name=cidr-block,Values=172.31.254.0/24 | jq '.Subnets | length')
+  if [[ "$match_count" != "1" ]]; then
+	  echo $(red "To run these tests you need to create a subnet with name \"isolated\" and CIDR \"172.31.254/24\"")
+	  exit 1
+  fi
+  	
+  if_count=$(aws ec2 describe-network-interfaces --filters Name=tag:nic-type,Values=hotpluggable | jq '.NetworkInterfaces[] | select(.["PrivateIpAddress"] | startswith("172.31.254.")) | .NetworkInterfaceId')
+  if [[ -z "$if_count" ]]; then
+	  echo $(red "To run these tests you need to create a network interface on the isolated subnet (CIDR \"172.31.254/24\") and tag it with \"nic-type:hotpluggable\"")
+	  exit 1
+  fi
+}
+
+
+# ec2_add_multi_nic_machine()
+#
+# Create a new machine, wait for it to boot and hotplug a pre-allocated
+# network interface which has been tagged: "nic-type: hotpluggable".
+#
+# Then, patch the netplan settings for the new interface, apply the new plan,
+# restart the machine agent and wait for juju to detect the new interface 
+# before returning.
+ec2_add_multi_nic_machine() {
+  juju add-machine
+  juju_machine_id=$(juju show-machine --format json | jq -r '.["machines"] | keys[0]')
+  echo "[+] waiting for machine ${juju_machine_id} to start..."
+  
+  until [ $(juju show-machine --format json | jq -r ".[\"machines\"] | .[\"${juju_machine_id}\"] | .[\"juju-status\"] | .[\"current\"]" | grep "started") ]; do
+    sleep 5
+  done
+
+  # Hotplug the second network device to the machine
+  echo "[+] hotplugging second NIC to machine ${juju_machine_id}..."
+  aws ec2 attach-network-interface --device-index 1 \
+  	--network-interface-id $(aws ec2 describe-network-interfaces --filters Name=tag:nic-type,Values=hotpluggable | jq -r '.NetworkInterfaces[0].NetworkInterfaceId') \
+  	--instance-id $(juju show-machine --format json | jq -r ".[\"machines\"] | .[\"${juju_machine_id}\"] | .[\"instance-id\"]")
+  
+  # Add an entry to netplan and apply it so the second interface comes online
+  echo "[+] updating netplan and restarting machine agent"
+  juju ssh ${juju_machine_id} 'sudo sh -c "echo \"            gateway4: `ip route | grep default | cut -d\" \" -f3`\n        ens6:\n            dhcp4: true\n\" >> /etc/netplan/50-cloud-init.yaml"'
+  juju ssh ${juju_machine_id} 'sudo netplan apply'
+  juju ssh ${juju_machine_id} 'sudo systemctl restart jujud-machine-*'
+  
+  # Wait for the interface to be detected by juju
+  echo "[+] waiting for juju to detect added NIC"
+  until [ $(juju show-machine --format json | jq -r ".[\"machines\"] | .[\"${juju_machine_id}\"] | .[\"network-interfaces\"] | length" | grep "3") ]; do
```

Comment:
Can we abstract this out similar to `wait_for`, I'm thinking `wait_for_machine`?

Reply:
Done; also indented output for all wait_for funcs